fv3config/filesystem.py

```diff
@@ -41,15 +40,24 @@ def is_existing_absolute_path(path: str) -> bool:
     return isabs(path) and get_fs(path).exists(path)
 
 
+class _Location:
+    def __init__(self, location: str) -> None:
+        self.match = re.match(r"(?P<prefix>(?P<protocol>.*?)://)?", location)
+        if self.match is None:
+            return f"protocol could not be inferred from {location}"
```

Comment:
`__init__` returning something other than None is confusing. Do we want to raise an exception here?

Reply:
yes. good catch. I meant to raise an exception.